keras/backend/tensorflow_backend.py

```diff
@@ -1284,6 +1284,8 @@ def _step(time, output_ta_t, *states):
                 output, new_states = step_function(current_input,
                                                    tuple(states) +
                                                    tuple(constants))
+                for state, new_state in zip(states, new_states):
+                    new_state.set_shape(state.get_shape())
```

Comment:
Surely this should be fixed inside your step function. When do you see issues arising?